simd/CMakeLists.txt

```diff
@@ -265,16 +265,15 @@ endif()
 # MIPS (GAS)
 ###############################################################################
 
-elseif(CPU_TYPE STREQUAL "mips" OR CPU_TYPE STREQUAL "mipsel")
+elseif(CPU_TYPE STREQUAL "mips" OR CPU_TYPE STREQUAL "mipsel" OR CPU_TYPE STREQUAL "mips64" OR CPU_TYPE STREQUAL "mips64el")
 
 enable_language(ASM)
 
 string(TOUPPER ${CMAKE_BUILD_TYPE} CMAKE_BUILD_TYPE_UC)
 set(EFFECTIVE_ASM_FLAGS "${CMAKE_ASM_FLAGS} ${CMAKE_ASM_FLAGS_${CMAKE_BUILD_TYPE_UC}}")
 message(STATUS "CMAKE_ASM_FLAGS = ${EFFECTIVE_ASM_FLAGS}")
 
-set(CMAKE_REQUIRED_FLAGS -mdspr2)
-
+set(CMAKE_REQUIRED_FLAGS "${CMAKE_C_FLAGS}")
```

Comment:
As we add MSA optimization mechanism for MIPS SIMD function, in order to be compatible with previous DSPR2 optimization mechanism I do this change. It depends on CMAKE_C_FLAGS to determine which one should be used:  If "-mdspr2" option is set and related c source checking can run passed, then SIMD optimization use DSPR2; otherwise "-mmsa" option is set and related c source checking can run passed, MSA optimization is used this time.

Reply:
OK, I will change this code later to keep DSPr2 detection mechanism as before and only add our MSA detection code to determine whether to use MSA optimization code or not.